ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/utils.py

```diff
@@ -167,11 +167,15 @@ def _resolve_p2p_ips(self, p2p_link: dict) -> dict:
         else:
             # Resolving subnet from pool
             ip_pool = get_item(self._p2p_links_ip_pools, "name", p2p_link["ip_pool"], default={}).get("ipv4_pool")
+            pool_prefix = get_item(self._p2p_links_ip_pools, "name", p2p_link["ip_pool"], default={}).get("ipv4_pool_mask")
+            if pool_prefix is None:
+                pool_prefix = 31
+
             if not ip_pool:
                 # Not possible to resolve from pool. Returning original
                 return p2p_link
             id = int(p2p_link["id"])
-            subnet = list(islice(ip_network(ip_pool).subnets(new_prefix=31), id - 1, id))[0]
+            subnet = list(islice(ip_network(ip_pool).subnets(new_prefix=int(pool_prefix)), id - 1, id))[0]
```

Comment:
Let's simplify this:
```suggestion
            # Resolving subnet from pool
            ip_pool = get_item(self._p2p_links_ip_pools, "name", p2p_link["ip_pool"], default={})
            ip_pool_subnet = ip_pool.get("ipv4_pool")
            if not ip_pool_subnet:
                return p2p_link
            pool_prefix = int(ip_pool.get("ipv4_pool_mask", 31))
            id = int(p2p_link["id"])
            subnet = list(islice(ip_network(ip_pool_subnet).subnets(new_prefix=pool_prefix), id - 1, id))[0]

```

Reply:
updated with sugestions